modules/services/window-managers/i3-sway/sway.nix

```
@@ -290,6 +290,21 @@ let
     '' else
       "") + cfg.extraConfig);
 
+  # Validates the Sway configuration
+  checkSwayConfig =
+    pkgs.runCommandLocal "sway-config" { buildInputs = [ cfg.package ]; } ''
+      # We have to make sure the wrapper does not start a dbus session
+      export DBUS_SESSION_BUS_ADDRESS=1
+
+      # A zero exit code means Sway succesfully validated the configuration
+      $DRY_RUN_CMD sway --config ${configFile} --validate --debug || {
```

What does $DRY_RUN_CMD do here?

---

`$DRY_RUN_CMD` is the empty string when home-manager is not run in "dry-run" otherwise it expands to `echo`

---

The `DRY_RUN_CMD` variable is only available in the activation script. This is a package build so it would not see that variable. I think it would be best to remove it.
```suggestion
      sway --config ${configFile} --validate --debug || {
```